libcaf_core/caf/detail/rfc3629.hpp

```diff
@@ -22,6 +22,16 @@ class CAF_CORE_EXPORT rfc3629 {
   static bool valid(std::string_view str) noexcept {
     return valid(as_bytes(make_span(str)));
   }
+
+  /// Checks whether `bytes` is a valid UTF-8 string.
+  static auto validate(const_byte_span bytes) noexcept
+    -> std::pair<size_t, bool>;
```

Comment:
I don't think we use trailing return type anywhere else in the code base. So for consistency:

```suggestion
  static std::pair<size_t, bool> validate(const_byte_span bytes) noexcept;
```

Same applies to the other functions.